tutorials/raw/plot_40_visualize_raw.py

```diff
@@ -122,6 +124,33 @@
 
 raw.plot_psd(average=True)
 
+###############################################################################
+# The `~mne.io.Raw.plot_psd` method estimates the spectrum using the Welch
+# method, which is efficient for long data spans. MNE-Python also implements
+# multitaper spectrum estimation as part of the ``time_frequency`` submodule,
+# though this requires doing the plotting manually using matplotlib, and taking
+# care to handle the different channel types separately:
```

Comment:
In general doing `psd_multitaper` on a raw instance can be quite slow. Even for `sample` which is just ~300 sec long it takes 12 sec to compute, and yields 83,000+ frequency points, which is... a lot. Should probably at least `.. note::` or warn people here that it will take a long time.

I wonder if instead of doing this we should actually implement welch + multitaper. In principle it's really just `psd_multitaper` called on `make_fixed_length_epochs(raw)`. It's even slower than `psd_multitaper(raw)` but it should produce a nicer result. At some point I'll see what adding a `multitaper=True` argument to `plot_psd` is like...

Reply:
> In general doing `psd_multitaper` on a raw instance can be quite slow.

I know.  Personally I don't think this addition to the tutorial is all that valuable, but I figured it must have been in the original example for a reason.  I can remove it, or I can add a warning and throw in a `tmax` for good measure, so it's not doing the full 60 seconds. WDYT?

> I wonder if instead of doing this we should actually implement welch + multitaper. In principle it's really just `psd_multitaper` called on `make_fixed_length_epochs(raw)`. It's even slower than `psd_multitaper(raw)` but it should produce a nicer result. At some point I'll see what adding a `multitaper=True` argument to `plot_psd` is like...

Personally I'd like to see `psd_welch` and `psd_multitaper` both return a `Spectrum` class that has its own plotting method(s). Then add a `raw.to_spectrum(method='welch'|'multitaper')` method that returns that `Spectrum` object. Then you could do `raw.to_spectrum().plot()`.  But that's a lot of work :)

Reply:
> I can remove it, or I can add a warning and throw in a tmax for good measure, so it's not doing the full 60 seconds. WDYT?

For now maybe just remove it. We should just improve `plot_psd` instead.

> > I wonder if instead of doing this we should actually implement welch + multitaper. 
> Then add a raw.to_spectrum(method='welch'|'multitaper') method

This misses the idea that you can combine welch and multitaper, though. I'll try to make a PR to show it. Then again I'm not sure if in practice people actually do combine them, it's even slower than either one alone anyway :)

Reply:
> This misses the idea that you can combine welch and multitaper, though. I'll try to make a PR to show it. Then again I'm not sure if in practice people actually do combine them, it's even slower than either one alone anyway :)

yes, sorry --- I understood the idea of combining them, I was just going off on one of my "how the API could be better" tangents. :)

I'll remove the addition now.